src/services/product/product.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Product } from '@entities/product';
+import { CreateProductDto, UpdateProductDto } from '@dtos/product';
+import {
+  ConflictException,
+  Injectable,
+  NotFoundException
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { ProductRepository } from '@repository/product';
+
+@Injectable()
+export class ProductService {
+  constructor(
+    @InjectRepository(ProductRepository)
+    private productRepository: ProductRepository
+  ) {}
+
+  async createProduct(creadteProductDto: CreateProductDto): Promise<Product> {
+    const product = await this.productRepository.createProduct(
+      creadteProductDto
+    );
+
+    return product;
+  }
+
+  async updateProduct(
+    id: string,
+    updateProductDto: UpdateProductDto
+  ): Promise<Product> {
+    const storedProduct = await this.productRepository.findOneBy({ id });
+    if (!storedProduct) {
+      throw new NotFoundException(`Product with id: ${id} was not found`);
+    }
+
+    const { name } = updateProductDto;
+
+    if (name !== storedProduct.name) {
```

Comment:
this logic here is kind of tricky, this service is used by the controller for the PUT operation that is supposed to edit a product given its `id`, but here you could end changing a product with a different id than the one provided that matches the name. In my opinion you should split this 2 cases to make them more explicit.